### PR TITLE
[FEAT] Home Widget for Favorites Album

### PR DESCRIPTION
**Pull request by prateekmedia**
## Description

This PR adds home widget support for both Android for favorites album.

The refresh time of widget slideshow is 10 minutes after which a different random image would be chosen.

Output:

iOS is blocked by workmanager plugin as it doesn't officially supports periodic refresh, but once it does it can be included again by reverting this commit: 96ca7cef81a6057abeb2262b140d2740c92b3f48

Solves #1471

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -61,14 +67,128 @@ const kBGPushTimeout = Duration(seconds: 28);
...
+}
+
 void main() async {
   debugRepaintRainbowEnabled = false;
   WidgetsFlutterBinding.ensureInitialized();
   MediaKit.ensureInitialized();
+
+  if (Platform.isAndroid) {
+    try {
+      await Workmanager()
```
Comment (ashilkn) [inline lib/main.dart:172]: Let's not await on this in `main()` and instead use `.then()` and `.onError()` on the futures? Don't want to increase the time app stays on splash screen.

File: lib/main.dart [hunk 2]
```diff
@@ -61,14 +67,128 @@ const kBGPushTimeout = Duration(seconds: 28);
...
+          .initialize(initSlideshowWidget, isInDebugMode: kDebugMode);
+
+      await Workmanager().registerPeriodicTask(
```
Comment (ashilkn) [inline lib/main.dart:175]: ^^

File: lib/ui/tabs/home_widget.dart [hunk 3]
```diff
@@ -60,6 +60,8 @@ import 'package:receive_sharing_intent/receive_sharing_intent.dart';
 import "package:shared_preferences/shared_preferences.dart";
 import 'package:uni_links/uni_links.dart';
 
+final scaffoldKey = GlobalKey<ScaffoldState>();
```
Comment (ashilkn) [inline lib/ui/tabs/home_widget.dart:63]: Don't see this used anywhere, lets remove?

File: lib/ui/tabs/home_widget.dart [hunk 4]
```diff
@@ -120,11 +122,13 @@ class _HomeWidgetState extends State<HomeWidget> {
         debugPrint(
           "TabChange going from $_selectedTabIndex to ${event.selectedIndex} souce: ${event.source}",
         );
-        _pageController.animateToPage(
-          event.selectedIndex,
-          duration: const Duration(milliseconds: 100),
-          curve: Curves.easeIn,
-        );
+        if (_pageController.hasClients) {
```
Comment (ashilkn) [inline lib/ui/tabs/home_widget.dart:125]: Why do we need this check?

Comment (prateekmedia) [reply lib/ui/tabs/home_widget.dart:125]: It was throwing errors if you turn on Uncaught Exceptions and run photos app.

File: lib/main.dart (hunk 1 above)
Comment (prateekmedia) [reply lib/main.dart]: Done! Although there is analysis check that requires await so I have used unawaited to fix this.

File: lib/ui/tabs/home_widget.dart (hunk 4 above)
Comment (ashilkn) [reply lib/ui/tabs/home_widget.dart]: After checking, this exception seems to be coming from `FileSelectionOverlayBar` and not `HomeWidget` so removing this check.

**Review by ashilkn — DISMISSED** [review]

File: lib/ui/tabs/home_widget.dart (hunk 4 above)
Comment (ashilkn) [reply lib/ui/tabs/home_widget.dart:125]: Update: @prateekmedia found that this is required for a different exception, so reverting removal of check.

**Review by ashilkn — APPROVED** [review]

**Pull request merged**